p2p/protocol/identify/id.go

```diff
@@ -894,8 +913,12 @@ func (nn *netNotifiee) Disconnected(_ network.Network, c network.Conn) {
 
 	// Stop tracking the connection.
 	ids.connsMu.Lock()
+	e, ok := ids.conns[c]
 	delete(ids.conns, c)
 	ids.connsMu.Unlock()
+	if ok && ids.metricsTracer != nil {
+		ids.metricsTracer.DecrementPushSupport(e.PushSupport)
```

Comment:
This could be removed if we just count newly identified connection, as suggested above.